src/app-layer-register.c

```diff
@@ -171,6 +171,11 @@ int AppLayerRegisterParser(const struct AppLayerParser *p, AppProto alproto)
                 p->GetTxIterator);
     }
 
+    if (p->SetTxDetectFlags && p->GetTxDetectFlags) {
```

Comment:
do we check anywhere for the case where only one of them would be set?

Reply:
No. Adding a `BOTH_SET_OR_BOTH_UNSET` check to `ValidateParserProto` to catch the case where only one is set.